packages/server/src/server.ts

```diff
@@ -202,17 +211,38 @@ export class Server extends EventEmitter {
     // Invoke the handler for the method requested
     const handler = this.handlers[req.method as keyof Handlers]
     if (handler) {
-      return handler.send(req, res).catch(onError)
+      return handler.send(req, res, context).catch(onError)
     }
 
-    return this.write(res, 404, 'Not found\n')
+    return this.write(context, req, res, 404, 'Not found\n')
   }
 
-  write(res: http.ServerResponse, status: number, body = '', headers = {}) {
+  write(
+    context: CancellationContext,
+    req: http.IncomingMessage,
+    res: http.ServerResponse,
+    status: number,
+    body = '',
+    headers = {}
+  ) {
+    const isAborted = context.signal.aborted
+
     if (status !== 204) {
       // @ts-expect-error not explicitly typed but possible
       headers['Content-Length'] = Buffer.byteLength(body, 'utf8')
     }
+
+    if (isAborted) {
+      // @ts-expect-error not explicitly typed but possible
+      headers['Connection'] = 'close'
```

Comment:
Please add a comment explaining what setting this header does. Does it instruct the http module to close the connection?

Reply:
This header instruct the client that we are willing to close the connection and to not send anymore data